src/tests/ftest/util/daos_utils.py

```diff
@@ -621,3 +636,116 @@ def pool_list_cont(self, pool, svc, sys_name=None, env=None):
         self.env = env
         return self._get_result()
+
+    def pool_set_attr(self, pool, attr, value, svc, env=None):
+        """Set pool attribute.
+
+        Args:
+            pool (String): Pool UUID.
+            attr (String): Attribute name.
+            value (String): Attribute value
+            svc (String): Service replicas.
+            env (dict, optional): dictionary of environment variable names and
+                values (EnvironmentVariables). Defaults to None.
+
+        Returns:
+            CmdResult: Object that contains exit status, stdout, and other
+                information.
+
+        Raises:
+            CommandFailure: if the doas pool query command fails.
+        """
+        self.set_sub_command("pool")
+        self.sub_command_class.set_sub_command("set-attr")
+        self.sub_command_class.sub_command_class.pool.value = pool
+        self.sub_command_class.sub_command_class.attr.value = attr
+        self.sub_command_class.sub_command_class.value.value = value
+        self.sub_command_class.sub_command_class.svc.value = svc
+        self.env = env
+        return self._get_result()
+
+    def pool_get_attr(self, pool, attr, svc, env=None):
+        """Set pool attribute.
+
+        Args:
+            pool (String): Pool UUID.
+            attr (String): Pool UUID.
+            svc (String): Service replicas.
+            env (dict, optional): dictionary of environment variable names and
+                values (EnvironmentVariables). Defaults to None.
+
+        Returns:
+            CmdResult: Object that contains exit status, stdout, and other
+                information.
+
+        Raises:
+            CommandFailure: if the doas pool query command fails.
+        """
+        self.set_sub_command("pool")
+        self.sub_command_class.set_sub_command("get-attr")
+        self.sub_command_class.sub_command_class.pool.value = pool
+        self.sub_command_class.sub_command_class.attr.value = attr
+        self.sub_command_class.sub_command_class.svc.value = svc
+        self.env = env
+        return self._get_result()
+
+    def pool_list_attrs(self, pool, svc, env=None):
+        """List pool attributes.
+
+        Args:
+            pool (String): Pool UUID.
+            svc (String): Service replicas.
+            env (dict, optional): dictionary of environment variable names and
+                values (EnvironmentVariables). Defaults to None.
+
+        Returns:
+            CmdResult: Object that contains exit status, stdout, and other
+                information.
+
+        Raises:
+            CommandFailure: if the doas pool query command fails.
+        """
+        self.set_sub_command("pool")
+        self.sub_command_class.set_sub_command("list-attrs")
+        self.sub_command_class.sub_command_class.pool.value = pool
+        self.sub_command_class.sub_command_class.svc.value = svc
+        self.env = env
+        return self._get_result()
+
+def get_pool_uuid(query_stdout):
```

Comment:
Same as the comment above, I would consider using `daos_cmd.get_ouput("pool_query", **kwargs)` in the test and then this function and the `get_sizes` function can be removed.

Reply:
I don't quite understand. What regex should I use? Should I try to get both UUID and the two size numbers with a single call of re.findall?

Reply:
I'm using a very similar regex for the test I have for the dmg pool_query command, since the ouput is so similar. (You can check it out in this PR: https://github.com/daos-stack/daos/pull/1674 in the get_pool_query_info function)
I modified it a bit for the daos_cmd output, the regex would be:
`(?:Pool\s*([A-Za-z0-9-]+),\s*ntarget=([0-9]),\s*disabled=([0-9])|Target\(VOS\) count:\s*([0-9])|(?:SCM:\s+.*|NVMe:\s+.*)Total\s+size:\s+([0-9]+)\s+.*Free:\s+([0-9]+),\s+min:([0-9]+),\s+max:([0-9]+),\s+mean:([0-9]+)|Rebuild\s*idle,\s*([0-9]+)\s*objs,\s*([0-9]+)\s*recs)`

Which would yield the following using `re.findall()`:

`[('3e59b386-fda0-404e-af7e-3ff0a38d1f81', '8', '0', '', '', '', '', '', '', '', ''), ('', '', '', '8', '', '', '', '', '', '', ''), ('', '', '', '', '1000000000', '999997440', '124999680', '124999680', '124999680', '', ''), ('', '', '', '', '0', '0', '0', '0', '0', '', ''), ('', '', '', '', '', '', '', '', '', '0', '0')]`

So, you would have to add the regex into the daos_utils.py.METHOD_REGEX dictionary. And from there, you can do the following in your test to get the list shown above:
```
kwargs = {"pool": pool_uuid}
daos_pool_query_out = self.get_output("pool_query", **kwargs)
``` 

In the function get_pool_query_info() in PR1674, I parse the array to take out all the empty string values and put the info into a dictionary. If you want to use that method, then you can get both the uuid info and the size info from the dictionary.

Reply:
I agree with Amanda.  The intent of the common `get_output()` method was to prevent the need for extra functions like `get_pool_uuid()` and `get_sizes()`

Reply:
Thanks for the regex. It worked like a charm. I only needed UUID and the two sizes, so I directly grabbed them from the findall output. I removed `get_pool_uuid()` and `get_sizes()`